src/interactive/results.ts

```diff
@@ -123,85 +162,240 @@ export class Result {
+            gotoPreviousFrame(frameArg.frame)
+        }),
+        vscode.commands.registerCommand('language-julia.gotoNextFrame', (frameArg: { frame: Frame }) => {
+            gotoNextFrame(frameArg.frame)
+        }),
+        vscode.commands.registerCommand('language-julia.gotoLastFrame', gotoLastFrame),
+        vscode.commands.registerCommand('language-julia.clearStackTrace', clearStackTrace)
+    )
 }
 
 export function deactivate() { }
 
-export function addResult(editor: vscode.TextEditor, range: vscode.Range, content: ResultContent) {
-    for (let i = results.length - 1; i > -1; i--) {
-        const result = results[i]
-        if (result.document === editor.document && result.range.intersection(range) !== undefined) {
-            removeResult(result)
-        }
+export function addResult(
+    editor: vscode.TextEditor,
+    range: vscode.Range,
+    content: string,
+    hoverContent: string
+) {
+    results.filter(result => result.document === editor.document && result.range.intersection(range) !== undefined).forEach(removeResult)
+    const result = new Result(editor, range, resultContent(content, hoverContent))
+    results.push(result)
+    return result
+}
+
+export function resultContent(content: string, hoverContent: string, isError: boolean = false): ResultContent {
+    return {
+        isIcon: false,
+        content,
+        hoverContent: toMarkdownString(hoverContent),
+        type: ResultType.Result,
+        isError
     }
+}
 
-    const result = new Result(editor, range, content)
-    results.push(result)
+const commandString = (cmd: string, args: string = '') => `command:${cmd}?${encodeURIComponent(args)}`
 
-    return result
+function toMarkdownString(str: string) {
+    const markdownString = new vscode.MarkdownString(str)
+    markdownString.isTrusted = true
+    return markdownString
+}
+
+export interface Frame {
+    path: string,
+    line: number
+}
+interface Highlight {
+    frame: Frame,
+    result: null | Result
+}
+
+interface StackFrameHighlights {
+    highlights: Highlight[]
+    err: string
+}
+
+const stackFrameHighlights: StackFrameHighlights = { highlights: [], err: '' }
+
+export function setStackTrace(result: Result, err: string, frames: Frame[]) {
+    clearStackTrace()
+    setStackFrameHighlight(err, frames)
+
+    result.onDidRemove(() => clearStackTrace())
+}
+
+export function clearStackTrace() {
+    stackFrameHighlights.highlights.forEach(highlight => {
+        if (highlight.result) {
+            highlight.result.remove()
+        }
+    })
+    stackFrameHighlights.highlights = []
+    stackFrameHighlights.err = ''
+}
+
+function setStackFrameHighlight(
+    err: string,
+    frames: Frame[],
+    editors: vscode.TextEditor[] = vscode.window.visibleTextEditors
+) {
+    stackFrameHighlights.err = err
+    frames.forEach(frame => {
+        const targetEditors = editors.filter(editor => frame.path === editor.document.fileName)
```

Comment:
This comparison doesn't work properly on Windows: Julia returns path with a capital drive letter, VS Code with a lower case drive letter, and we never get any match here.

We should probably in general base all of this on URIs, not on paths, and try to use the VS Code URI type as much as we can for this kind of thing. See [here](https://code.visualstudio.com/updates/v1_46#_file-path-casing) for some recent casing specific points.

Reply:
Fixed now. Turns out we need both this comparison (for untitled editors) _and_ a comparison of the normalized URIs (for normal files).

Reply:
Can't we do all the comparisons based on URIs? The URI for an unsaved file is `untitled:Untitled-1`, i.e. it doesn't use the `file:` scheme (because it isn't), but the `untitled:` scheme instead. But I guess the problem is that we can't know whether a given file name we get in a stack trace is a file that exists or an untitled file, right?

Reply:
Yup. We can of course have a heuristic that constructs the URI with the right scheme when the file name starts with `Untitled`, but meh...